lib/form/js/validation.js

```diff
@@ -6,90 +6,100 @@
   'use strict';
 
   var valueCheck = function valueCheck(input, settings, validation) {
-    if (input.target.required && (input.target.required === 'save' || input.target.required === 'publish') && input.target.value === '') {
-      return 'Field cannot be left blank!';
-    }
-
-    return validation(input, settings);
-  };
+        if (input.target.required && (input.target.required === 'save' || input.target.required === 'publish') && input.target.value === '') {
+          return 'Field cannot be left blank!';
+        }
 
-  // Validation Function
-  // Disabled one-var for better readability
-  var validate = function validate(e) { // eslint-disable-line one-var
-    var target = e.target,
-        value = target.value,
-        inp = allInputs[target.id],
-        parent = allIDs[inp.parent.id],
-        input = {},
-        settings = {},
-        validation = false,
-        message,
-        currentMessage = parent.querySelector('[role="alert"][for="' + target.id + '"]'),
-        insertRef;
+        return validation(input, settings);
+      },
 
-    // Set up validation input
-    input.target = {
-      value: value,
-      name: inp.name,
-      required: inp.required,
-    };
-    input.all = {};
-    input.all[inp.name] = value;
+      // Validation Function
+      validate = function validate(e) {
+        var target = e.target,
+            value = target.value,
+            inp = allInputs[target.id],
+            parent = target,
+            input = {},
+            settings = {},
+            validation = false,
+            message,
+            currentMessage = parent.querySelector('[role="alert"][for="' + target.id + '"]'),
+            insertRef;
 
-    // Set up validation settings
-    settings.target = inp.settings;
-    settings.all = {};
-    settings.all[inp.name] = inp.settings;
+        while (parent.className.indexOf('form--repeatable') === -1 &&
```

Comment:
A couple of thoughts on this:

* Instead of using `className.indexOf` (which would be `true` for something like `bar-form--repeatable` which is not what we want), use `classList.contains('form--repeatable')` as that will tell you if the actual class is there instead of if it exists in a string
* Instead of just removing `closest` all together, check to see if the functionality exists. If it does exist, use it, otherwise fall back to the while loop. It's a little more code, but it means that for browsers that support that emerging standard, they'll be able to use a native method (which will be faster)